src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionClassTypeParameterNameTest.java

```diff
@@ -0,0 +1,128 @@
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+///////////////////////////////////////////////////////////////////////////////////////////////
+
+package org.checkstyle.suppressionxpathfilter;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import com.puppycrawl.tools.checkstyle.DefaultConfiguration;
+import com.puppycrawl.tools.checkstyle.checks.naming.AbstractClassNameCheck;
+import com.puppycrawl.tools.checkstyle.checks.naming.ClassTypeParameterNameCheck;
+
+public class XpathRegressionClassTypeParameterNameTest extends AbstractXpathTestSupport {
+
+    private final String checkName = ClassTypeParameterNameCheck.class.getSimpleName();
+
+    @Override
+    protected String getCheckName() {
+        return checkName;
+    }
+
+    @Test
+    public void testDefault() throws Exception {
+        final File fileToProcess =
+                new File(getPath("InputXpathClassTypeParameterNameDefault.java"));
+        final String pattern = "^[A-Z]$";
+        final DefaultConfiguration moduleConfig =
+                createModuleConfig(ClassTypeParameterNameCheck.class);
+
+        final String[] expectedViolation = {
+            "5:20: " + getCheckMessage(ClassTypeParameterNameCheck.class,
+                        AbstractClassNameCheck.MSG_ILLEGAL_ABSTRACT_CLASS_NAME, "abc", pattern),
+        };
+
+        final List<String> expectedXpathQueries = Arrays.asList(
+                "/COMPILATION_UNIT/CLASS_DEF"
+                        + "[./IDENT[@text='InputXpathClassTypeParameterNameDefault']]"
+                        + "/OBJBLOCK/CLASS_DEF[./IDENT[@text='MyClass2']]"
+                        + "/TYPE_PARAMETERS/TYPE_PARAMETER[./IDENT[@text='abc']]",
+                "/COMPILATION_UNIT/CLASS_DEF"
+                        + "[./IDENT[@text='InputXpathClassTypeParameterNameDefault']]"
+                        + "/OBJBLOCK/CLASS_DEF[./IDENT[@text='MyClass2']]/TYPE_PARAMETERS"
+                        + "/TYPE_PARAMETER/IDENT[@text='abc']"
+        );
+
+        runVerifications(moduleConfig, fileToProcess, expectedViolation,
+                expectedXpathQueries);
+    }
+
+    @Test
+    public void testUpperCase() throws Exception {
+        final String pattern = "^[A-Z]{2,}$";
+        final File fileToProcess =
+                new File(getPath("InputXpathClassTypeParameterNameUpperCase.java"));
```

Comment:
Please update input file names and test methods to reflect structure content file, not a specificof name of class.